backend/book-app/src/main/java/com/karankumar/bookproject/account/controller/UserController.java

```diff
@@ -88,6 +90,16 @@ public User getUser(@PathVariable Long id) {
         );
   }
 
+  @GetMapping("/email/{email}")
+  public ResponseEntity<String> checkIfEmailExists(@PathVariable String email) {
+      Optional<User> user = userService.findUserByEmail(email);
```

Comment:
```
userService.findUserByEmail(email)
    .map(user-> ResponseEntity.ok())
    .orElse(ResponseEntity.status(HttpStatus.NOT_FOUND).body(EMAIL_NOT_FOUND));
 ```

Reply:
Thanks, @mslowiak! That's a lot more concise!